templates/16_update.tpl

```diff
@@ -157,12 +165,23 @@ func (o {{$tableNameSingular}}Slice) UpdateAll(exec boil.Executor, cols M) error
 		pkeyArgs := queries.ValuesFromMapping(reflect.Indirect(reflect.ValueOf(obj)), {{$varNameSingular}}PrimaryKeyMapping)
 		args = append(args, pkeyArgs...)
 	}
-
+	{{if ne .DriverName "mssql" -}}
 	sql := fmt.Sprintf(
 		"UPDATE {{$schemaTable}} SET %s WHERE ({{.LQ}}{{.Table.PKey.Columns | join (printf "%s,%s" .LQ .RQ)}}{{.RQ}}) IN (%s)",
 		strmangle.SetParamNames("{{.LQ}}", "{{.RQ}}", {{if .Dialect.IndexPlaceholders}}1{{else}}0{{end}}, colNames),
 		strmangle.Placeholders(dialect.IndexPlaceholders, len(o) * len({{$varNameSingular}}PrimaryKeyColumns), len(colNames)+1, len({{$varNameSingular}}PrimaryKeyColumns)),
 	)
+	{{- else -}}
+	startIndex := len(colNames) + 1
+
+	if !dialect.IndexPlaceholders {
+		startIndex = 0
+	}
+
+	sql := fmt.Sprintf("UPDATE {{$schemaTable}} SET %s WHERE %s",
+		strmangle.SetParamNames("{{.LQ}}", "{{.RQ}}", {{if .Dialect.IndexPlaceholders}}1{{else}}0{{end}}, colNames),
+		strmangle.WhereClauseRepeated(string(dialect.LQ), string(dialect.RQ), startIndex, {{$varNameSingular}}PrimaryKeyColumns, len(o)))
```

Comment:
Okay. Done talking. We talked about the maintenance burden of having to ever touch these queries, and updating them in 6 places instead of 3 sort of thing, and setting a precedent for other database engines. If we ever were to add sqlite we don't want to have yet another huge amount of sections saying: "if sqlite", we tried to limit it to exceptional cases for mysql and we should do the same for mssql.

So we should simply use the bottom query (the one you wrote for MSSQL) for all database engines (at least I can't see a reason it wouldn't work for the others) and so I think we should do that. And we should do the same in both `18_delete.tpl` and `19_reload.tpl`.

Also one change to these queries is they're doing an if at runtime to choose startindex, if you see the original query that's done at template time and it should continue to be that way. If it's not runtime dependent it should always be done by templates.